cargo-shuttle/tests/integration/init.rs

```diff
@@ -2,17 +2,22 @@ use std::fs::read_to_string;
 use std::path::Path;
 use std::process::Command;
 
-use cargo_shuttle::{Args, Shuttle};
+use cargo_shuttle::{Shuttle, ShuttleArgs};
 use clap::Parser;
 use indoc::indoc;
 use tempfile::Builder;
 
+// quite high timeout since the template is being cloned over network
+const EXPECT_TIMEOUT_MS: u64 = 10000;
+
 #[tokio::test]
 async fn non_interactive_basic_init() {
     let temp_dir = Builder::new().prefix("basic-init").tempdir().unwrap();
+    // Sleep to give time for the directory to finish creating
+    tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
```

Comment:
I was getting 'no such directoy' error when debugging sporadic failing tests. This was an attempt to track them down, but I left them in to be sure. The tests' run duration don't suffer from this.
I think the errors happen internally in cargo-generate, they also create and delete tmp dirs.

I can remove the sleeps if wanted but they don't hurt either. CI has improved a good bit on this branch.

Reply:
We might be able to avoid the sleeps by working with a global tempdir, that gets dropped after all tests run. They run in serial order and the `ScopedWorkingDirectory` Drop impl seems only to rely on setting the `env::current_dir` when dropped. Not sure why is this.. but hopefully it will not affect how the testing goes. What do you think?

Reply:
Strange! I see a warning about early drops of the temp_dir in the tempfile docs, but it doesn't seem like `.path().to_owned()` will drop it. This is a bit of a shot in the dark, but could it be because several of the tests use the same name for the temp dir? Either way I think we can merge with these sleeps, but it would be good to be certain that they are required.